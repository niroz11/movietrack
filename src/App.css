
.movie-section{
  overflow: scroll;
}
.nav-bar{
  display: flex;
  justify-items: center;
  align-items: center;
  width:15%;
  justify-content: space-between;
}

.App-header{
  display: flex;
  justify-content: space-between;
}

.heading-title{
  width:85%;
  justify-items: center;
  align-items: center;
  margin-bottom: 30px;
  margin-top:20px;
}
.App{
  background-color: #1f1f1f;
}
.movie-box-heading{
  color: #fcfcfc;
  padding-left: 10px;
  margin: 0;
}
.nav-bar a {
  color:#f1f1f1;
  text-decoration: none;
  font-size: 20px;
}
.nav-bar a:hover{
  text-decoration: underline;

}








