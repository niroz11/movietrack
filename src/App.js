import React, { Component } from 'react'
import { Route, NavLink } from 'react-router-dom'
import { Login } from './Components/Login/Login'
import  Nowplayingmovies  from './Components/Nowplayingmovies/Nowplayingmovies'
import Toprated from './Components/Toprated/Toprated'
import Popularmovies from './Components/Popularmovies/Popularmovies'
import { APIkey } from './utils/APIkey'
import { fetchOptions } from './utils/fetchOptions'
import { fetchData } from './utils/fetchData'
import { connect } from 'react-redux'
import { nowPlayingMovies, topRatedMovies, popularMovies } from './actions/actions'


class App extends Component {
  constructor(){
    super()
    this.state={
      error:''
    }
  }

  componentDidMount = () => {
    this.fetchNowPlayingMovies()
    this.fetchTopRatedMovies()
    this.fetchPopularMovies()
  }

  fetchNowPlayingMovies = async () => {
    try{
    const options = await fetchOptions('GET')
    const movies = await fetchData(APIkey,options)
    this.props.nowPlayingMovies(movies.results)
    } catch(error){
      this.setState({error: error.message})
    }
  }

  fetchTopRatedMovies = async() => {
    const url = "https://api.themoviedb.org/3/movie/top_rated?api_key=93b214404de014118b64ce033e70ac99&language=en-US&page=1"
    try{
      const options = await fetchOptions('GET')
      const movies = await fetchData(url,options)
      this.props.topRatedMovies(movies.results)
      

    }catch(error){
      this.setState({
        error: error.message
      })
    }
  }

  fetchPopularMovies = async() => {
    const url = "https://api.themoviedb.org/3/movie/popular?api_key=93b214404de014118b64ce033e70ac99&language=en-US&page=1"
    try{
      const options = await fetchOptions('GET')
      const movies = await fetchData(url,options)
      this.props.popularMovies(movies.results)
      

    }catch(error){
      this.setState({
        error: error.message
      })
    }
  }
   
  render() {
    console.log(this.props, "app props")
    return (
      <div className="App">
        <header className="App-header">
          <div className="heading-title">
            <h1>MovieTrack</h1>
            
          </div>
          <div>
            <NavLink to="/Login" className="login-button">Login</NavLink>
            <NavLink to="/signup" className ="signup-button">Signup</NavLink>
            <Route path='/Login' component = {Login}/>
            <Route exact path='/Nowplaying' component={Nowplayingmovies}/>
          </div>
        </header>
        <section className="movies-container">
         <div className="now-playing movie-section">
         <h1>Now playing movies</h1>
            <Nowplayingmovies/>
         </div>
         <div className="toprated-movies movie-section">
         <h1>Top rated movies</h1>
          <Toprated/>
         </div>
         <div className="popular-movies movie-section">
         <h1>Popular movies</h1>
         <Popularmovies/>
         </div>
        
        </section>
        
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  nowPlayingMovies: (movies) => dispatch(nowPlayingMovies(movies)),
  topRatedMovies: (movies) => dispatch(topRatedMovies(movies)),
  popularMovies: (movies) => dispatch(popularMovies(movies))
  
})

export default connect(null, mapDispatchToProps)(App)
