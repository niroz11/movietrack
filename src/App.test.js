import React from 'react';
import ReactDOM from 'react-dom';
import { App }  from './App';
import { shallow, mount } from 'enzyme';
import { fetchNowPlayingMovies } from './App';

describe('App', () => {
  let wrapper;
  const mockMovies =  [
    {title: "Cars", poster_path: "car.jpg"},
    {title: "How To Train Your Dragon", poster_path: "dragon.jpg"}
  ]
    
  const mockNowPlayingMovies = jest.fn()
  const mockTopRatedMovies = jest.fn()
  const mockPopularMovies = jest.fn()
    // const mockFailedRequest = jest.fn().mockRejectedValue(new Error('I failed'));
  
  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      ok: true, 
      json: () => Promise.resolve({
        results: mockMovies
      })
    }))
    wrapper = shallow(<App nowPlayingMovies={mockNowPlayingMovies} 
                           topratedMovies={mockTopRatedMovies} 
                           popularMovies={mockPopularMovies}
                           user={{id:1, name: "taylor"}}
                           />)
  })

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call all of my fetches when the component mounts', () => {
    wrapper.instance().fetchNowPlayingMovies = jest.fn()
    wrapper.instance().fetchTopRatedMovies = jest.fn()
    wrapper.instance().fetchPopularMovies = jest.fn()
    wrapper.instance().componentDidMount()
    expect(wrapper.instance().fetchNowPlayingMovies).toHaveBeenCalled()
    expect(wrapper.instance().fetchTopRatedMovies).toHaveBeenCalled()
    expect(wrapper.instance().fetchPopularMovies).toHaveBeenCalled()

  })

  it('should set an error state when the fetch fails', async () => {
    // First, I want to reassign window.fetch to reject so that I have an error
    // Next, I want to make sure that fetch gets called with this new definition so that I have an error
    // Finally, I want to expect that my component's error state is equal to the error message
    
    window.fetch = jest.fn().mockImplementationOnce(() => Promise.reject(
      new Error('failed')
    ))
    await wrapper.instance().fetchNowPlayingMovies()
    expect(wrapper.state("error")).toEqual("failed")



  })

  it('should use fetchNowPlayingMovies to call fetch', async () => {
      expect(window.fetch).toHaveBeenCalled()
  })
  it('should dispatch nowPlayingMovies with mock movies', async () => {
    wrapper = shallow(<App nowPlayingMovies={mockNowPlayingMovies} 
      topratedMovies={mockTopRatedMovies} 
      popularMovies={mockPopularMovies}
      user={{id:1, name: "taylor"}}
      />)
    await wrapper.instance().fetchNowPlayingMovies()
    
    expect(wrapper.instance().props.nowPlayingMovies).toHaveBeenCalledWith(mockMovies)
  
    
  })
});
