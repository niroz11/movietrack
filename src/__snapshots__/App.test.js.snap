// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    nowPlayingMovies={[MockFunction]}
    popularMovies={[MockFunction]}
    topratedMovies={[MockFunction]}
    user={
      Object {
        "id": 1,
        "name": "taylor",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <div
            className="heading-title"
          >
            <h1>
              MovieTrack
            </h1>
          </div>
          <div>
            <NavLink
              className="login-button"
              to="/logout"
            >
              Logout
            </NavLink>
            <NavLink
              className="signup-button"
              to="/signup"
            >
              Signup
            </NavLink>
            <NavLink
              className="favorites-button"
              to="/favorites"
            >
              Favorites
            </NavLink>
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/favorites"
            />
          </div>
        </header>,
        <section
          className="movies-container"
        >
          <Route
            exact={true}
            path="/movie/:id"
            render={[Function]}
          />
          <div
            className="now-playing movie-section"
          >
            <h1>
              Now playing movies
            </h1>
            <Connect(Nowplayingmovies) />
          </div>
          <div
            className="toprated-movies movie-section"
          >
            <h1>
              Top rated movies
            </h1>
            <Connect(Toprated) />
          </div>
          <div
            className="popular-movies movie-section"
          >
            <h1>
              Popular movies
            </h1>
            <Connect(Popularmovies) />
          </div>
        </section>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="heading-title"
            >
              <h1>
                MovieTrack
              </h1>
            </div>,
            <div>
              <NavLink
                className="login-button"
                to="/logout"
              >
                Logout
              </NavLink>
              <NavLink
                className="signup-button"
                to="/signup"
              >
                Signup
              </NavLink>
              <NavLink
                className="favorites-button"
                to="/favorites"
              >
                Favorites
              </NavLink>
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
              />
            </div>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                MovieTrack
              </h1>,
              "className": "heading-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieTrack",
              },
              "ref": null,
              "rendered": "MovieTrack",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  className="login-button"
                  to="/logout"
                >
                  Logout
                </NavLink>,
                <NavLink
                  className="signup-button"
                  to="/signup"
                >
                  Signup
                </NavLink>,
                <NavLink
                  className="favorites-button"
                  to="/favorites"
                >
                  Favorites
                </NavLink>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/signup"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/favorites"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Logout",
                  "className": "login-button",
                  "onClick": undefined,
                  "to": "/logout",
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Signup",
                  "className": "signup-button",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "Signup",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Favorites",
                  "className": "favorites-button",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/movie/:id"
              render={[Function]}
            />,
            <div
              className="now-playing movie-section"
            >
              <h1>
                Now playing movies
              </h1>
              <Connect(Nowplayingmovies) />
            </div>,
            <div
              className="toprated-movies movie-section"
            >
              <h1>
                Top rated movies
              </h1>
              <Connect(Toprated) />
            </div>,
            <div
              className="popular-movies movie-section"
            >
              <h1>
                Popular movies
              </h1>
              <Connect(Popularmovies) />
            </div>,
          ],
          "className": "movies-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/movie/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Now playing movies
                </h1>,
                <Connect(Nowplayingmovies) />,
              ],
              "className": "now-playing movie-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Now playing movies",
                },
                "ref": null,
                "rendered": "Now playing movies",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Top rated movies
                </h1>,
                <Connect(Toprated) />,
              ],
              "className": "toprated-movies movie-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Top rated movies",
                },
                "ref": null,
                "rendered": "Top rated movies",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Popular movies
                </h1>,
                <Connect(Popularmovies) />,
              ],
              "className": "popular-movies movie-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Popular movies",
                },
                "ref": null,
                "rendered": "Popular movies",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <div
              className="heading-title"
            >
              <h1>
                MovieTrack
              </h1>
            </div>
            <div>
              <NavLink
                className="login-button"
                to="/logout"
              >
                Logout
              </NavLink>
              <NavLink
                className="signup-button"
                to="/signup"
              >
                Signup
              </NavLink>
              <NavLink
                className="favorites-button"
                to="/favorites"
              >
                Favorites
              </NavLink>
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/signup"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
              />
            </div>
          </header>,
          <section
            className="movies-container"
          >
            <Route
              exact={true}
              path="/movie/:id"
              render={[Function]}
            />
            <div
              className="now-playing movie-section"
            >
              <h1>
                Now playing movies
              </h1>
              <Connect(Nowplayingmovies) />
            </div>
            <div
              className="toprated-movies movie-section"
            >
              <h1>
                Top rated movies
              </h1>
              <Connect(Toprated) />
            </div>
            <div
              className="popular-movies movie-section"
            >
              <h1>
                Popular movies
              </h1>
              <Connect(Popularmovies) />
            </div>
          </section>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="heading-title"
              >
                <h1>
                  MovieTrack
                </h1>
              </div>,
              <div>
                <NavLink
                  className="login-button"
                  to="/logout"
                >
                  Logout
                </NavLink>
                <NavLink
                  className="signup-button"
                  to="/signup"
                >
                  Signup
                </NavLink>
                <NavLink
                  className="favorites-button"
                  to="/favorites"
                >
                  Favorites
                </NavLink>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/signup"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/favorites"
                />
              </div>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  MovieTrack
                </h1>,
                "className": "heading-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "MovieTrack",
                },
                "ref": null,
                "rendered": "MovieTrack",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    className="login-button"
                    to="/logout"
                  >
                    Logout
                  </NavLink>,
                  <NavLink
                    className="signup-button"
                    to="/signup"
                  >
                    Signup
                  </NavLink>,
                  <NavLink
                    className="favorites-button"
                    to="/favorites"
                  >
                    Favorites
                  </NavLink>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/signup"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/favorites"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Logout",
                    "className": "login-button",
                    "onClick": undefined,
                    "to": "/logout",
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Signup",
                    "className": "signup-button",
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": "Signup",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Favorites",
                    "className": "favorites-button",
                    "to": "/favorites",
                  },
                  "ref": null,
                  "rendered": "Favorites",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/signup",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/favorites",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/movie/:id"
                render={[Function]}
              />,
              <div
                className="now-playing movie-section"
              >
                <h1>
                  Now playing movies
                </h1>
                <Connect(Nowplayingmovies) />
              </div>,
              <div
                className="toprated-movies movie-section"
              >
                <h1>
                  Top rated movies
                </h1>
                <Connect(Toprated) />
              </div>,
              <div
                className="popular-movies movie-section"
              >
                <h1>
                  Popular movies
                </h1>
                <Connect(Popularmovies) />
              </div>,
            ],
            "className": "movies-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/movie/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Now playing movies
                  </h1>,
                  <Connect(Nowplayingmovies) />,
                ],
                "className": "now-playing movie-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Now playing movies",
                  },
                  "ref": null,
                  "rendered": "Now playing movies",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Top rated movies
                  </h1>,
                  <Connect(Toprated) />,
                ],
                "className": "toprated-movies movie-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Top rated movies",
                  },
                  "ref": null,
                  "rendered": "Top rated movies",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Popular movies
                  </h1>,
                  <Connect(Popularmovies) />,
                ],
                "className": "popular-movies movie-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Popular movies",
                  },
                  "ref": null,
                  "rendered": "Popular movies",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
